pkg/hintrunner/zero/zerohint_keccak.go

```diff
@@ -1,24 +1,105 @@
 package zero
 
 import (
+	"fmt"
 	"math"
 
 	"github.com/NethermindEth/cairo-vm-go/pkg/hintrunner/hinter"
 	VM "github.com/NethermindEth/cairo-vm-go/pkg/vm"
+	"github.com/NethermindEth/cairo-vm-go/pkg/vm/builtins"
+	"github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
 	mem "github.com/NethermindEth/cairo-vm-go/pkg/vm/memory"
 	"github.com/consensys/gnark-crypto/ecc/stark-curve/fp"
 	"github.com/holiman/uint256"
 )
 
+// CairoKeccakFinalize writes the result of F1600 Keccak permutation padded by __keccak_state_size_felts__ zeros to consecutive memory cells, __block_size__ times.
+//
+// `CairoKeccakFinalize` takes 3 operanders as arguments
+//   - `keccakStateSizeFelts` is the number of felts in the Keccak state size
+//   - `blockSize` is the number of blocks to write
+//   - `keccakPtrEnd` is the address in memory where to start writing the result
+//
+// The `low` and `high` parts are splitted in 64-bit integers
+// Ultimately, the result is written into 4 memory cells
+func newCairoKeccakFinalizeHint(keccakStateSizeFelts, blockSize, keccakPtrEnd hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "CairoKeccakFinalize",
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			//> _keccak_state_size_felts = int(ids.KECCAK_STATE_SIZE_FELTS)
+			//> _block_size = int(ids.BLOCK_SIZE)
+			//> assert 0 <= _keccak_state_size_felts < 100
+			//> assert 0 <= _block_size < 10
+			//> inp = [0] * _keccak_state_size_felts
+			//> padding = (inp + keccak_func(inp)) * _block_size
+			//> segments.write_arg(ids.keccak_ptr_end, padding)
+
+			keccakStateSizeFeltsVal, err := hinter.ResolveAsUint64(vm, keccakStateSizeFelts)
+			if err != nil {
+				return err
+			}
+			if keccakStateSizeFeltsVal >= 100 {
+				return fmt.Errorf("assertion failed: 0 <= keccak_state_size_felts < 100.")
+			}
+			blockSizeVal, err := hinter.ResolveAsUint64(vm, blockSize)
+			if err != nil {
+				return err
+			}
+			if blockSizeVal >= 10 {
+				return fmt.Errorf("assertion failed: 0 <= _block_size < 10.")
+			}
+
+			var input [25]uint64
+			builtins.KeccakF1600(&input)
+			padding := make([]uint64, keccakStateSizeFeltsVal)
+			padding = append(padding, input[:]...)
+			result := make([]uint64, 0, keccakStateSizeFeltsVal*blockSizeVal)
```

Comment:
shouldn't the capacity be `(keccakStateSizeFeltsVal + len(input)) * blockSizeVal`

Reply:
`padding = (inp + keccak_func(inp)) * _block_size` 

` keccak_func(inp)` 's size is the same as `inp` 

so i'm thinking the capacity should be `keccakStateSizeFeltsVal * 2` instead of `keccakStateSizeFeltsVal * blockSizeVal` ? 

for each iteration around blockSizeVal, we need to write to 2 *  keccakStateSizeFeltsVal memory cells, no?